src/ServerRequestFactory.php

```diff
@@ -59,14 +77,17 @@ public function createFromParameters(array $server, array $headers = [], array $
     {
         $method = $server['REQUEST_METHOD'] ?? null;
         if ($method === null) {
-            throw new \RuntimeException('Unable to determine HTTP request method.');
+            throw new RuntimeException('Unable to determine HTTP request method.');
         }
 
         $uri = $this->getUri($server);
-
         $request = $this->serverRequestFactory->createServerRequest($method, $uri, $server);
 
         foreach ($headers as $name => $value) {
+            if ($name === 'Host' && $request->hasHeader('Host')) {
+                continue;
```

Comment:
```suggestion
            if ($name === 'Host') {
                $request->withHeader('Host', $value)
                continue;
```
Do you think, depending on the implementation of the ResponseInterface, could a different result be achieved?
If yes, what would be the best case?

Reply:
PSR implementations must add `Host` and make it the first one.
https://tools.ietf.org/html/rfc7230#section-5.4

If we do as you suggest, the header will always be overwritten and `Host` will not be the first one.

Reply:
> PSR implementations must add Host and make it the first one.
> https://tools.ietf.org/html/rfc7230#section-5.4

This requirement applies ONLY to the text format of the HTTP request. In object representation (PSR implementations), it all depends on the implementation of the MessageInterface.

You should follow this:
https://www.php-fig.org/psr/psr-7/#host-header

Reply:
In this case, the title will be set if it is not there!

This is also necessary when emiting headers.